src/aggregator/aggregator/generic_elem.go

```diff
@@ -293,10 +294,8 @@ func (e *GenericElem) expireValuesWithLock(
 	}
 	resolution := e.sp.Resolution().Window
 
-	// start after the minimum to ensure we always keep at least one value in the map for binary transformations.
-	currStart := e.minStartTime.Add(resolution)
+	currStart := e.minStartTime
```

Comment:
we actually want to run this loop for `e.minStartTime` so we close the aggregation.